cli/tests/nonce.rs

```diff
@@ -53,6 +53,63 @@ fn test_nonce() {
     let (keypair_file, mut tmp_file) = make_tmp_file();
     write_keypair(&config_nonce.keypair, tmp_file.as_file_mut()).unwrap();
 
+    full_battery_tests(
+        &rpc_client,
+        &drone_addr,
+        &mut config_payer,
+        &mut config_nonce,
+        &keypair_file,
+        &keypair_file,
+    );
+
+    server.close().unwrap();
+    remove_dir_all(ledger_path).unwrap();
+}
+
+#[test]
+fn test_nonce_with_authority() {
+    let (server, leader_data, alice, ledger_path) = new_validator_for_tests();
+    let (sender, receiver) = channel();
+    run_local_drone(alice, sender, None);
+    let drone_addr = receiver.recv().unwrap();
+
+    let rpc_client = RpcClient::new_socket(leader_data.rpc);
+
+    let mut config_payer = CliConfig::default();
+    config_payer.json_rpc_url =
+        format!("http://{}:{}", leader_data.rpc.ip(), leader_data.rpc.port());
+
+    let mut config_nonce = CliConfig::default();
+    config_nonce.json_rpc_url =
+        format!("http://{}:{}", leader_data.rpc.ip(), leader_data.rpc.port());
+    let (nonce_keypair_file, mut tmp_file) = make_tmp_file();
+    write_keypair(&config_nonce.keypair, tmp_file.as_file_mut()).unwrap();
+
+    let nonce_authority = Keypair::new();
+    let (authority_keypair_file, mut tmp_file2) = make_tmp_file();
+    write_keypair(&nonce_authority, tmp_file2.as_file_mut()).unwrap();
+
+    full_battery_tests(
+        &rpc_client,
+        &drone_addr,
+        &mut config_payer,
+        &mut config_nonce,
+        &nonce_keypair_file,
+        &authority_keypair_file,
+    );
+
+    server.close().unwrap();
+    remove_dir_all(ledger_path).unwrap();
+}
+
+fn full_battery_tests(
```

Comment:
full battery?

Reply:
"any large group or series of related things"